### PR TITLE
Add_classes

### PR DESCRIPTION
**Pull request by NalianAchei2020**
✅ Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config). 
✅ Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
 ✅ Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

> Ruby requirements 
✅ Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

> Project requirements Create class Person with the following: 
✅ Instance vars: @id, @name, and @Age. 
✅ Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have 
✅ default values of "Unknown" and true.
 ✅ Getters for @id, @name, and @Age. 
✅ Setters for @name and @Age.
 ✅ Private method is_of_age? that returns true if @Age is greater or equal to 18 and false otherwise. 
✅ Public method can_use_services? that returns true if person is of age or if they have permission from parents. Create class Student with the following: 
✅ Inherits from Person
 ✅ Constructor extends parent's constructor by adding @classroom and a parameter for it. 
✅ Method play_hooky that returns "¯(ツ)/¯". Create class Teacher with the following:
 ✅ Inherits from Person. 
✅ Constructor extends parent's constructor by adding https://github.com/specialization and a parameter for it.
 ✅ Override can_use_services? so it always returns true. Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+  # getter and setter in one
+  attr_accessor :name
+  attr_accessor :age
+
+  # getter
+  attr_reader :id
```
Comment (Deepakdanger) [inline person.rb]: - [ ] `:name & :age` can be use on same line, for ex: `attr_accessor :name, :age `.
- [ ] Kindly move the getter & setter before initiaze method, on line 2, Always use getter & setter at top (conventionally)

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+  private :is_of_age?
+
+  def can_use_services?
+    @parent_permission || is_of_age?
+  end
```
Comment (Deepakdanger) [inline person.rb]: - [ ] Here line 18 & 21 , used `is_of_age?` but defined method is `of_age?`, kindly correct it.
- [ ] [OPTIONAL] there is also good way to use the private method like this 

```ruby
  def can_use_services? 
    of_age?() || @parent_permission
  end
  private 
  
  def of_age?
    @age >= 18
  end

```
when used private like this all method written below `private` is private method, and above one are public.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @NalianAchei2020 , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 ✔️ 

Hi @NalianAchei2020, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] All requested changes are made ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**